src/OpenSage.Game.Tests/DataStructures/QuadtreeTests.cs

```diff
@@ -135,5 +157,293 @@ public void InsertAndRemoveNonExistant()
+
+        [Fact]
+        public void RectangleIntersectsOthersOverlappingBoundary()
+        {
+            var quadtree = new Quadtree<MockQuadtreeItem>(new RectangleF(0, 0, 10, 10));
+
+            var item = new MockQuadtreeItem(1, new RectangleF(0, 0, 1, 1));
+            var item2 = new MockQuadtreeItem(1, new RectangleF(0, 0, 1, 1));
+
+            quadtree.Insert(item);
+            quadtree.Insert(item2);
+
+            var item3 = new MockQuadtreeItem(1, new RectangleF(4, 0, 2, 2));
+            quadtree.Insert(item3);
+
+            Assert.Equal(new[] { item3 }, quadtree.FindIntersecting(new BoxCollider(new RectangleF(5, 0, 2, 2))));
+            Assert.Equal(new[] { item3 }, quadtree.FindIntersecting(new BoxCollider(new RectangleF(3, 0, 2, 2))));
+        }
+
+        #endregion
+
+        // This tests a large number of items against the "dumb" way of checking for collisions to ensure the quadtree implementation is correct.
+        [Fact]
+        public void FuzzFindIntersecting()
+        {
+            const int itemsToGenerate = 1000;
+            const float minX = 0f;
+            const float minY = 0f;
+            const float maxX = 200f;
+            const float maxY = 150f;
+            const int seed = 42; // fix seed to fuzz results are consistent
+
+            var r = new Random(seed);
+
+            var quadtree = new Quadtree<MockQuadtreeItem>(new RectangleF(minX, minY, maxX, maxY));
+
+            var items = new List<MockQuadtreeItem>();
+
+            for (var i = 0; i < itemsToGenerate; i++)
+            {
+                var iMinX = minX + r.NextDouble() * (maxX - minX);
+                var iMinY = minY + r.NextDouble() * (maxY - minY);
+                var width = r.NextDouble() * Math.Min(maxX - iMinX, maxX / 10);
+                var height = r.NextDouble() * Math.Min(maxY - iMinY, maxY / 10);
+
+                var item = new MockQuadtreeItem(i, new RectangleF((float)iMinX, (float)iMinY, (float)width, (float)height));
+                items.Add(item);
+                Assert.True(quadtree.Insert(item));
+            }
+
+            var collisions = new Dictionary<int, HashSet<int>>();
+
+            // build a manual register of intersecting items to test the quadtree against
+            foreach (var item1 in items)
+            {
+                if (!collisions.ContainsKey(item1.Id))
+                {
+                    collisions[item1.Id] = [];
+                }
+
+                foreach (var item2 in items.Where(item2 => item1.RoughCollider.Intersects(item2.RoughCollider)))
+                {
+                    if (item1.Id == item2.Id)
+                    {
+                        continue;
+                    }
+
+                    if (!collisions.ContainsKey(item2.Id))
+                    {
+                        collisions[item2.Id] = [];
+                    }
+
+                    collisions[item1.Id].Add(item2.Id);
+                    collisions[item2.Id].Add(item1.Id);
```

Comment:
If nothing else, we should make sure this is correct. As long as this is correct, then if the fuzzing passes we can feel confident that our quadtree implementation is correct.